### PR TITLE
Create more specific BC files for BlameReports

### PR DESCRIPTION
**Pull request by vulder**
To improve analysis precision BlameReports need TBAA metadata, which
only get's added to clang at opt level 01. Hence, we need to disable
optimisations via an extra clang arg. These special requirements on BC
files are encoded as new extensions.

Furthermore, this commit adds an ordering to the BCFileExtension class,
so multiple BCFileExtension can be sorted before creating the file name.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

File: tests/data/test_wllvm.py [hunk 1]
```diff
@@ -5,6 +5,24 @@
 from varats.experiment.wllvm import Extract, BCFileExtensions
 
 
+class TestBCFileExtensions(unittest.TestCase):
+    """Test if the Extension enum is correctly setup."""
+
+    def test_ordering(self):
+        """Checks if the extension are ordered/sorted correclty."""
```
Comment (cormensratio) [inline tests/data/test_wllvm.py]:
```suggestion
        """Checks if the extension are ordered/sorted correctly."""
```

File: varats-core/varats/experiment/wllvm.py [hunk 2]
```diff
@@ -30,9 +30,25 @@
 
 
 class BCFileExtensions(Enum):
+    """
+    List of possible extensions that specify the way a BC file was created.
+
+    An extension should be requested when a BC file needs to fullfill certain
```
Comment (cormensratio) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
    An extension should be requested when a BC file needs to fulfill certain
```

File: tests/data/test_wllvm.py [hunk 3]
```diff
@@ -5,6 +5,24 @@
...
+        exts = [
+            BCFileExtensions.NO_OPT, BCFileExtensions.TBAA,
+            BCFileExtensions.DEBUG
+        ]
+
+        self.assertEqual(
+            sorted(exts), [
+                BCFileExtensions.NO_OPT, BCFileExtensions.TBAA,
```
Comment (cormensratio) [inline tests/data/test_wllvm.py:20]: Don't we need the BCFileExtensions.OPT also included in the test?

**Review by cormensratio — APPROVED** [review]
lgtm, just two typos and one question.

File: tests/data/test_wllvm.py (hunk 3 above)
Comment (vulder) [reply tests/data/test_wllvm.py:20]: We don't need all of them here. Just a few to see if sorting works.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

File: tests/data/test_wllvm.py [hunk 4]
```diff
@@ -5,6 +5,24 @@
 from varats.experiment.wllvm import Extract, BCFileExtensions
 
 
+class TestBCFileExtensions(unittest.TestCase):
+    """Test if the Extension enum is correctly setup."""
+
+    def test_ordering(self):
+        """Checks if the extension are ordered/sorted correctly."""
```
Comment (boehmseb) [inline tests/data/test_wllvm.py]:
```suggestion
        """Checks if the extensions are ordered/sorted correctly."""
```

File: tests/data/test_wllvm.py [hunk 5]
```diff
@@ -5,6 +5,24 @@
 from varats.experiment.wllvm import Extract, BCFileExtensions
 
 
+class TestBCFileExtensions(unittest.TestCase):
+    """Test if the Extension enum is correctly setup."""
```
Comment (boehmseb) [inline tests/data/test_wllvm.py]:
```suggestion
    """Test if the Extension enum is correctly set up."""
```

**Review by boehmseb — APPROVED** [review]
lgtm

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=6270452&bid=20412577)

**Pull request merged**